services/web/server/src/simcore_service_webserver/wallets/_payments_handlers.py

```diff
@@ -0,0 +1,164 @@
+
+_logger = logging.getLogger(__name__)
+
+
+routes = web.RouteTableDef()
+
+
+def _raise_if_not_dev_mode(app):
+    app_settings = get_settings(app)
+    if not app_settings.WEBSERVER_DEV_FEATURES_ENABLED:
+        msg = "This feature is only available in development mode"
+        raise NotImplementedError(msg)
+
+
+async def _fake_delay_for_processing_time():
+    # Fakes processing time
+    processing_time = random.uniform(0.5, 2)  # noqa: S311 # NOSONAR #nosec
+    await asyncio.sleep(processing_time)
+
+
+async def _fake_payment_completion(app: web.Application, payment: WalletPaymentCreated):
+    # NOTE: tmp fakes payment completion for the front-end
+    await _fake_delay_for_processing_time()
+
+    # Three different possible outcomes
+    possible_outcomes = [
+        # 1. Accepted
+        {"app": app, "payment_id": payment.payment_id, "success": True},
+        # 2. Rejected
+        {
+            "app": app,
+            "payment_id": payment.payment_id,
+            "success": False,
+            "message": "Payment rejected",
+        },
+        # TODO: 3. does not complete ever
+    ]
+    kwargs: dict[str, Any] = random.choice(possible_outcomes)  # noqa: S311
+    await complete_payment(**kwargs)
+
+
+@routes.post(f"/{VTAG}/wallets/{{wallet_id}}/payments", name="create_payment")
+@login_required
+@permission_required("wallets.*")
+@handle_wallets_exceptions
+async def create_payment(request: web.Request):
+    req_ctx = WalletsRequestContext.parse_obj(request)
+    path_params = parse_request_path_parameters_as(WalletsPathParams, request)
+    body_params = await parse_request_body_as(CreateWalletPayment, request)
+
+    _raise_if_not_dev_mode(request.app)
+
+    # ensure the wallet can be used by the user
+    wallet: WalletGet = await get_wallet_by_user(
+        request.app,
+        user_id=req_ctx.user_id,
+        wallet_id=path_params.wallet_id,
+        has_write_permission=True,  # Can only pay to wallets that user owns
+    )
+    wallet_id = wallet.wallet_id
+
+    with log_context(
+        _logger,
+        logging.INFO,
+        "Payment transaction started to %s",
+        f"{wallet_id=}",
+        log_duration=True,
+        extra=get_log_record_extra(user_id=req_ctx.user_id),
+    ):
+        payment = await create_payment_to_wallet(
+            request.app,
+            user_id=req_ctx.user_id,
+            product_name=req_ctx.product_name,
+            wallet_id=wallet.wallet_id,
+            wallet_name=wallet.name,
+            osparc_credit=body_params.osparc_credits,
+            price_dollars=body_params.price_dollars,
+            comment=body_params.comment,
+        )
+
+    # NOTE: fake completion for the moment
+    await fire_and_forget_task(
+        _fake_payment_completion(request.app, payment),
+        task_suffix_name=f"fake_payment_completion_{payment.payment_id}",
+        fire_and_forget_tasks_collection=request.app[APP_FIRE_AND_FORGET_TASKS_KEY],
+    )
```

Comment:
If I get this correctly, this will not be the final pattern. This block of code will go away, right?

Reply:
Please beware that in case of a of the webserver the payment will never be completed.

Reply:
The more I think about this the less I like it. The current tests are biased to this line of code. Can't the frontend emulate this if you require it for testing? Can't we have a different endpoint for it.

Reply:
moved as a separate event